SPDY/NSURLRequest+SPDYURLRequest.m

```diff
@@ -154,4 +188,20 @@ - (void)setSPDYBodyFile:(NSString *)SPDYBodyFile
     }
 }
 
+- (void)setSPDYURLSession:(NSURLSession *)SPDYURLSession
+{
+    [self spdy_setIndirectObject:SPDYURLSession forKey:@"SPDYURLSession"];
+}
+
+- (void)spdy_setIndirectObject:(id)object forKey:(NSString *)key
+{
+    if (object == nil) {
+        [SPDYProtocol removePropertyForKey:key inRequest:self];
```

Comment:
you still need to call objc_setAssociatedObject with the nil object in this condition as well. Otherwise the runtime will continue to hold onto the previous object until a new non-nil object is set for this key.